### PR TITLE
Implement conversion methods for Float types and add corresponding tests

### PR DESCRIPTION
**Pull request by shogo82148**
## Summary by CodeRabbit

- **New Features**
  - Introduced methods for converting between custom floating-point types (Float16, Float32, Float64, Float128, Float256).
  - Added a command-line utility for validating float32 to float64 conversions at the bit level.
- **Tests**
  - Added unit tests to verify conversion methods between Float32 and Float64 types, including edge cases for overflow and underflow.
- **Chores**
  - Added scripts to automate building of SoftFloat and TestFloat libraries.
  - Added a script to run floating-point tests with configurable seeds.
- **Maintenance**
  - Updated `.gitignore` to exclude test and binary directories.

### CODE REVIEW

File: convert_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		},
+		{
+			in:   2.0,
+			want: 2.0,
+		},
+	}
+
+	for _, tt := range tests {
+		if got := tt.in.Float32(); got != tt.want {
+			t.Errorf("Float32(%x).Float32() = %x, want %x", tt.in, got, tt.want)
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply convert_test.go:25]: Golang supports hexadecimal representation of floating-point numbers. This is displayed without any issues.
